packages/react/src/hooks/useVisibilityLoader.js

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import warning from 'warning';
+import { merge } from 'lodash-es';
+
+import { browserSupports } from '../utils/componentUtilityFunctions';
+
+const defaultOptions = {
+  /** are there more items available to load */
+  hasMoreToLoad: true,
+  /** is the application already loading new data */
+  isLoading: false,
+  /** a callback that will be triggered when the ref element becomes visible */
+  onVisible: () => {},
+  /* options passed to IntersectionObserver, see: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API */
+  intersectionObserverOptions: {
+    rootMargin: '100px 0px 100px 0px',
+  },
+};
+
+/**
+ * A simple hook to return a bool telling if the given element is visible on the screen. It can be
+ * used to load a single item or multiple items. If maxToLoad is greater than 1, the first item in
+ * the returned array will be an array of booleans telling how many of the items are to be loaded. The
+ * loadBatchSize prop determines how many new items to mark as visible each time the reference element
+ * is visible on screen. For example, if the loadBatchSize is 20 and you're scrolling down a table,
+ * when the reference row becomes visible it will update the isVisible array to contain 20 new true
+ * booleans.
+ *
+ * @param {*} ref html element to check if it's visible before rendering more items
+ * @param {object} options
+ * @returns Array an array containing [isVisible: bool | arrayOf(bool), moreToLoad: bool]
+ */
+const useVisibilityLoader = (ref, options) => {
+  if (!browserSupports('IntersectionObserver')) {
+    warning(
+      !__DEV__,
+      'The current browser does not support IntersectionObserver. You will need to include a IntersectionObserver polyfill for this component to function properly.'
```

Comment:
Since this is replacing the Visibility Sensor does the visibility sensor require the polyfill?  ie would this be a breaking change when this is applied?

Reply:
I thought I commented on this already, but I'm not seeing my response now, so I'm sorry if this is a duplicate. Yes, this will be a breaking change for browsers that don't support IntersectionObserver. I just check the internals of react-visibility-sensor and it's using onScroll detection. We can either lazy-load the polyfill here in this check or push this change to v3 to prevent the potential breaks.

Reply:
IntersectionObserver is supported in all modern browsers so this should just be a safeguard. 
![image](https://user-images.githubusercontent.com/1590966/145886311-2cd6d372-2647-49ad-90d1-27ec2e3f660b.png)